### ISSUE TITLE
BaseCompat account.status gives unknown error on 0.4.0 while code was working on production using old SDKs

### ISSUE THREAD

**Issue by l-hendriks** [kinecosystem/kin-ios#10]
Hello! I'm working on integrating the Agora SDK for KinFit, which is already running on kin for quite some time. I updated the kin-ios version in the podfile to 0.4.0, rebuilt and now the `account.status` function does not work anymore.

Here is the relevant part of the code (I added some asdasd so I can more easily find it in the logs):

```swift
        ...
        print("asdasdQ")
        print(self.account)
        self.account!.status { (status: AccountStatus?, error: Error?) in
            print("asdasdX")
            print(status)
            print(error)
            guard let status = status else { return }

            ...

```

Here are the logs:

```
asdasdQ
Optional(KinSDK.KinAccount)
2020-11-21 10:15:45.440887+0100 kinfit[66572:7024944] KinBase.KinAccountContext::info::getAccount(forceUpdate:)
2020-11-21 10:15:45.443145+0100 kinfit[66572:7024944] KinBase.KinService::debug::[Request][V3]====
CreateAccountRequest(accountId: "GAA7C3Z2ATHIFIVS2PNKT2ZJLDKP3TOPF5GZKO3QTTZZO2U35NIK3QHY")
=====[Request][V3]
2020-11-21 10:15:45.443657+0100 kinfit[66572:7024896] KinBase.AgoraGrpcProxy::debug::AgoraGrpcProxy::request::<APBAccountV3CreateAccountRequest 0x600002e081e0>: {
    account_id {
      value: "GAA7C3Z2ATHIFIVS2PNKT2ZJLDKP3TOPF5GZKO3QTTZZO2U35NIK3QHY"
    }
}
2020-11-21 10:15:45.997364+0100 kinfit[66572:7024944] KinBase.AgoraGrpcProxy::debug::AgoraGrpcProxy::response::<APBAccountV3CreateAccountResponse 0x600002e7ab80>: {
    result: EXISTS
    account_info {
      account_id {
        value: "GAA7C3Z2ATHIFIVS2PNKT2ZJLDKP3TOPF5GZKO3QTTZZO2U35NIK3QHY"
      }
      sequence_number: 40682033306927104
    }
}
2020-11-21 10:15:45.998102+0100 kinfit[66572:7024944] KinBase.KinService::debug::[Response][V3]====
CreateAccountResponse(result: KinBase.CreateAccountResponse.Result.exists, error: nil, account: nil)
=====[Response][V3]
asdasdX
nil
Optional(KinBase.KinService.Errors.unknown)
```

So as you see, everything is working nicely (Agora tells the account already exists), but when I do `account.status` I receive an `Optional(KinBase.KinService.Errors.unknown)`

Any ideas?

Thanks!

**Comment by l-hendriks** [comment]
Just checked, the issue is not resolved on 0.4.1

**Comment by chaseeb** [comment]
resolved
